pyfr/plugins/kernels/turbulence.mako

```diff
@@ -0,0 +1,51 @@
+<% import numpy as np %>
+
+<%namespace module='pyfr.backends.base.makoutil' name='pyfr'/>
+
+<%pyfr:macro name='turbulence' params='t, u, ploc, src'>
+  fpdtype_t arg, clip, g;
+  fpdtype_t tpos[3], tploc[3], delta2[3], utilde[3] = {};
+
+  uint32_t oldstate, newstate, rshift;
+
+  % for i, rotc in enumerate(rot):
+    tploc[${i}] = ${' + '.join(f'{r}*(ploc[{j}] - {s})' for j, (r, s) in enumerate(zip(rotc, shift)))};
+  % endfor
+
+  % for i in range(nvmx):
+    tpos[0] = ${-ls} + (t - tinit[${i}])*${avgu};
+
+    % for j in range(2):
+      oldstate = ${f'state[{i}]' if loop.index == 0 else 'newstate'};
+      newstate = oldstate * 747796405UL + 2891336453UL;
+      rshift = oldstate >> 28UL;
+      oldstate ^= oldstate >> (4UL + rshift);
+      oldstate *= 277803737UL;
+      oldstate ^= oldstate >> 22UL;
+      tpos[${j + 1}] = ${yzmin[j]} + ${yzdim[j]}*((fpdtype_t)(oldstate >> 8UL) * 5.9604644775390625e-8);
```

Comment:
Write the floating point number as a Mako expression to make its origin clearer.